File: 370a/main.py
import sys

input = lambda: sys.stdin.readline().rstrip()


def I():
    return input()


def II():
    return int(input())


def MII():
    return map(int, input().split())


def LI():
    return list(input().split())


def LII():
    return list(map(int, input().split()))

# alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
r1, c1, r2, c2 = MII()
rook = 1 if r1 == r2 or c1 == c2 else 2
bishop = 0
if (r1 - c1) % 2 == (r2 - c2) % 2:
    bishop = 1 if abs(r2 - r1) == abs(c2 - c1) else 2
king = max(abs(r1 - r2), abs(c1 - c2))
print(rook, bishop, king)

File: 370a/main.go
package main

import (
	"bufio"
	. "fmt"
	"io"
	"os"
)

func abs(x int) int {
	if x < 0 {
		return -x

	}
	return x
}

func run(r io.Reader, w io.Writer) {
	in := bufio.NewReader(r)
	out := bufio.NewWriter(w)
	defer out.Flush()
	var r1, c1, r2, c2 int
	Fscan(in, &r1, &c1, &r2, &c2)

	rook := 2
	if r1 == r2 || c1 == c2 {
		rook = 1
	}
	bishop := 0
	if (r1-c1)&1 == (r2-c2)&1 {
		bishop = 2
		if abs(r2-r1) == abs(c2-c1) {
			bishop = 1
		}
	}

	king := max(abs(r1-r2), abs(c1-c2))
	Fprintln(out, rook, bishop, king)

}

func main() {
	run(os.Stdin, os.Stdout)
}
